src/v/container/include/container/lw_shared_container.h

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2024 Redpanda Data, Inc.
+ *
+ * Use of this software is governed by the Business Source License
+ * included in the file licenses/BSL.md
+ *
+ * As of the Change Date specified in that file, in accordance with
+ * the Business Source License, use of this software will be governed
+ * by the Apache License, Version 2.0
+ */
+
+#include "base/seastarx.h"
+
+#include <seastar/core/shared_ptr.hh>
+
+#pragma once
+
+template<typename C>
+class lw_shared_container {
+public:
+    using iterator = C::iterator;
```

Comment:
since it's shared, is it worth making it const?

Reply:
Compiles as suggested, so I'm going to say :shipit: